app/views/search/search.html.haml

```diff
@@ -2,38 +2,29 @@
   %h1.label-heading
     Search
   = render partial: "search_form"
-
   - if @q
-    - if @scrapers.any? || @owners.any?
-      %ul.search-type-nav.nav.nav-tabs
-        %li{class: ('active' unless @type == "users"), role: "presentation"}
-          = link_to search_path(q: @q, type: nil) do
-            Scrapers
-            %span.badge #{@scrapers.total_count}
-        %li{class: ('active' if @type == "users"), role: "presentation"}
-          = link_to search_path(q: @q, type: "users") do
-            Users
-            %span.badge #{@owners.total_count}
-
-      - if @type == "users"
-        .search-results
-          - @owners.with_details.each do |owner, details|
-            = render owner, highlight: details[:highlight]
-        = paginate @owners
-      - else
-        .search-results
-          - @scrapers.with_details.each do |scraper, details|
-            = render scraper, highlight: details[:highlight]
-        = paginate @scrapers
-
-      - if @type == "users" && @owners.empty? && @scrapers.any?
-        .alert.alert-info
-          Sorry, we couldn’t find any users relevant to your search term
-          %strong “#{@q}”.
-      - elsif @type != "users" && @scrapers.empty? && @owners.any?
-        .alert.alert-info
-          Sorry, we couldn’t find any scrapers relevant to your search term
-          %strong “#{@q}”.
+    - if !@unfiltered_scrapers_count.zero? || !@owners.empty?
+      .search-results.row
+        .col-xs-12
+          = render partial: "search_type_nav"
+        - if @type == "users"
+          - if !@owners.total_count.zero?
+            .col-xs-12
+              = render partial: "search_results_list", locals: {results: @owners}
+          - else
+            = render partial: "no_results_of_type_notice"
+        - else
+          - if !@unfiltered_scrapers_count.zero?
+            %aside.col-xs-12.col-md-3
+              = render partial: "scraper_search_filter_controls"
+            - # If there are scrapers matching filters
+            - if !@scrapers.total_count.zero?
+              .col-xs-12.col-md-9
+                = render partial: "search_results_list", locals: {results: @scrapers}
+            - else
+              = render partial: "no_results_of_type_notice"
+          - else
+            = render partial: "no_results_of_type_notice"
```

Comment:
By splitting this out into a partial you've separated two bits that deal with similar things - which is what to show when there are no results. These are now in the partial "no_results_of_type_notice" and in the code below. General guiding principle: Similar things should be close together.